src/send-request.ts

```diff
@@ -29,7 +32,7 @@ export const addParamsToURL = (
 
 export const encodePostData = (data: PostData | Uint8Array, options: Partial<XHROptions>): string | BlobPart | null => {
     if (options.blob && data.buffer) {
-        return new Blob([data.buffer], { type: 'text/plain' })
+        return new Blob([_isUint8Array(data) ? data : data.buffer], { type: 'text/plain' })
```

Comment:
our JS tests were using snapshots here. And the snapshot was happily passing when this method returned the empty array for Uint8Arrays passed in